detect_secrets/core/usage.py

```diff
@@ -44,6 +65,36 @@ def add_no_verify_flag(parser):
     )
 
 
+def add_shared_arguments(parser):
+    """
+    These are arguments that are for both
+    pre-commit and regular detect-secrets scripts.
+    """
+    add_exclude_lines_argument(parser)
+    add_word_list_argument(parser)
+    add_custom_plugins_argument(parser)
+    add_use_all_plugins_argument(parser)
+    add_no_verify_flag(parser)
+
+
+def get_parser_to_add_opt_out_options_to(parser):
+    """
+    The pre-commit hook gets e.g. `--no-jwt-scan` type options
+    as well as the subparser for `detect-secrets scan`.
+
+    :rtype: argparse.ArgumentParser
+    :returns: argparse.ArgumentParser to pass into PluginOptions
+    """
+    if parser.prog == 'detect-secrets-hook':
+        return parser
+
+    for action in parser._actions:  # pragma: no cover (Always returns)
```

Comment:
Bottom answer of https://stackoverflow.com/questions/43688450/how-to-obtain-argparse-subparsers-from-a-parent-parser-to-inspect-defaults had some great pointers, works in both 2 and 3.